pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
And that (a single global struct) is a problem because.....? It's not like
this is a massive object memory wise.

As I said, we're already down that path(this struct contains content that
is not relevant to all build strategies), so either we should stick with it
(generic universal object) or we should be splitting the content we already
have into specialized structs.  Personally I prefer the single object
because
1) we are already doing that
2) if in the future we introduce a new build type that needs some
combination of fields that exist in other strategies or specialized
objects, it's going to be a pain again.

But going halfway does not make sense to me.

As for what dockerfile does, it is not used for source strategies. Hence my
point.

It is used to provide a dockerfile (in string form) for a docker strategy
build to use, either by itself, or in conjunction with a git repo that is
also provided. (and it will override a dockerfile that exists in that git
repo)

Ben Parees | OpenShift
On Nov 25, 2015 8:40 AM, "Maciej Szulik" <notifications@github.com> wrote:

> In pkg/build/api/v1/types.go
> <https://github.com/openshift/origin/pull/5976#discussion_r45866934>:
>
> > @@ -163,6 +163,9 @@ type BuildSource struct {
> >  	// repository.
> >  	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
> >
> > +	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
> > +	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
>
> @bparees <https://github.com/bparees> but going down your path we end up
> blowing up our BuildSource structure with data that are applicable only for
> certain strategies. That's why I always liked the strategy approach, where
> you define strategy-specific fields.
>
> Btw. I've wondered about the existence of Dockerfile string field in
> BuildSource, what it's for in the first place? If I'm looking correctly
> it's for overwriting source's Dockerfile
> <https://github.com/openshift/origin/blob/e66f133f88238b55441bab4772ed2456c72c7f01/pkg/build/builder/source.go#L60>,
> is that correct? What for? Isn't source repo enough for that? If so,
> shouldn't if be part of SourceBuildStrategy, especially that in here
> <https://github.com/openshift/origin/blob/e66f133f88238b55441bab4772ed2456c72c7f01/pkg/build/generator/generator.go#L375>
> we are clearing that field under certain conditions, which even more tells
> me it's not right.
> Sorry for those lame questions, but I wasn't following builds for a while
> and I'm kinda confused about those changes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5976/files#r45866934>.
>